writeups/2020-12-16-winter-contest-2020/index.md

```diff
@@ -0,0 +1,585 @@
+
+The last thing we need to sort out is the vtable pointer. The first function call after `read` is `puts` (actually it's the *only* function call, because `_exit` will exit the process immediately without running any of the usual exit code), which calls `_IO_xsputn_t` at offset `0x30` from the vtable. We want to call `_IO_wfile_sync` at `_IO_wfile_jumps+0x58`, so we will set the vtable to `_IO_wfile_jumps+0x58-0x30`.
+
+## Solving
+First, we send `-6` as the index and grab one of the libc addresses. We can use this to find the base address of libc, after which we can carry out the FSOP attack.
+
+Since we only get one write, the implementation is a bit tricky. Luckily, most of the struct members are irrelevant, so we can overlap all three of the structs.
+
+![final payload layout](./seegink/final.png)
+
+
+
+The left-most block is the final payload, and the members that each element corresponds to is highlighted in color. All the empty rows are null.
+
+Remember that we are reading starting at offset 2, so we will need to cut the first two bytes off the beginning. The last thing we need to watch out for is the final `strlen`. Fortunately, the first byte we write is a null, so `strlen` returns 0 and writes a 0 to `_flags`, which is what we need.
+
+
+## Flag
+```
+flag{stdout_m0r3_l1ke_stdsice_2e2c312e24316b3f2c35}
+```
+
+# admin-plz-v3
+
+> Are these admins ever gonna stop?
+> https://admin-plz-v3.winter-challenge.tjcsec.club/
+
+Written by saisree
+
+A continuation of the admin-plz series of challenges. If you hated it, well, here's good news: it's the last one. Or is it?
+
+## Inspection
+
+Looking at the site, we see a login form appear with a nice transition. Looking first at the source code, nothing seems to pop out immediately; just some CSS with a form sending a POST request to the url `/login`.
+
+If we enter in a random username or password, we are greeted with the message "Wrong username or password."
+
+From the challenge `admin-plz-v2`, we might guess that this login page contains a SQL Injection vulnerability. Let's try a standard trick where we input the username as `admin` and the password as `' OR '1'='1`:
+
+![SQLi filter](./admin/filter.jpg)
+
+Looks like there's some filtration here. Let's try substituting `||` for `OR`. Same result.
+
+We could try different techniques to bypass the filters, but it seems like the admins did a decent job with that, so we are a little stuck.
+
+Let's look at some other aspects of the site to see if we find anything. Looking at the network requests, nothing seems odd—just the POST request to the server. If we look at the cookies, however, there's something interesting: a cookie called `encoding`. This is its value:
+
+`YldRMUlHSjFkQ0J0WVd0bElITjFjbVVnZEc4Z1pHVnNaWFJsSUhSb2FYTWdiR0YwWlhJaElTRT0=`
```

Comment:
This, up until ``## Exploitation`` is overflowing the view on mobile, may need triple backtick

Reply:
smh head saigautam

Reply:
lmao